samples/BasicXrApp/DxUtility.cpp

```diff
@@ -100,21 +101,43 @@ namespace xr::dx {
         return compiled;
     }
 
-    // Create a list of feature levels which are both supported by the OpenXR runtime and this application.
-    std::vector<D3D_FEATURE_LEVEL> SelectFeatureLevels(D3D_FEATURE_LEVEL minimumFeatureLevel) {
-        // Create a list of feature levels which are both supported by the OpenXR runtime and this application.
-        std::vector<D3D_FEATURE_LEVEL> featureLevels = {D3D_FEATURE_LEVEL_12_1,
-                                                        D3D_FEATURE_LEVEL_12_0,
-                                                        D3D_FEATURE_LEVEL_11_1,
-                                                        D3D_FEATURE_LEVEL_11_0,
-                                                        D3D_FEATURE_LEVEL_10_1,
-                                                        D3D_FEATURE_LEVEL_10_0};
-        featureLevels.erase(std::remove_if(featureLevels.begin(),
-                                           featureLevels.end(),
-                                           [&](D3D_FEATURE_LEVEL fl) { return fl < minimumFeatureLevel; }),
-                            featureLevels.end());
-        CHECK_MSG(featureLevels.size() != 0, "Unsupported minimum feature level!");
-        return featureLevels;
+    SwapchainD3D11 CreateSwapchainD3D11(XrSession session,
+                                        DXGI_FORMAT format,
+                                        int32_t width,
+                                        int32_t height,
+                                        uint32_t arrayLength,
+                                        uint32_t sampleCount,
+                                        XrSwapchainCreateFlags createFlags,
+                                        XrSwapchainUsageFlags usageFlags) {
+        SwapchainD3D11 swapchain;
+        swapchain.Format = format;
+        swapchain.Width = width;
+        swapchain.Height = height;
+        swapchain.ArrayLength = arrayLength;
+
+        XrSwapchainCreateInfo swapchainCreateInfo{XR_TYPE_SWAPCHAIN_CREATE_INFO};
+        swapchainCreateInfo.arraySize = arrayLength;
+        swapchainCreateInfo.format = format;
+        swapchainCreateInfo.width = width;
+        swapchainCreateInfo.height = height;
+        swapchainCreateInfo.mipCount = 1;
+        swapchainCreateInfo.faceCount = 1;
+        swapchainCreateInfo.sampleCount = sampleCount;
+        swapchainCreateInfo.createFlags = createFlags;
+        swapchainCreateInfo.usageFlags = usageFlags;
+
+        CHECK_XRCMD(xrCreateSwapchain(session, &swapchainCreateInfo, swapchain.Handle.Put()));
```

Comment:
I think this function is better suited somewhere else. It is related to DX11, but it doesn't actually use it any. If the goal is to put all OpenXR function calls in OpenXrProgram.cpp then this is a small leak.

Reply:
Good point. for OpenXrProgram.cpp, it's better to move this.   When we have more samples, this function can be useful in shared lib, but that's later.